app/controllers/ideas_controller.rb

```diff
@@ -66,13 +67,22 @@ def submit
     end
   end
 
+  def current_user_has_voted_on_idea?
+    @vote.any?
+  end
+  helper_method :current_user_has_voted_on_idea?
+
   private
 
   # Use callbacks to share common setup or constraints between actions.
   def set_idea
     @idea = Idea.find(params[:id] || params[:idea_id])
```

Comment:
Probably worth changing this to `includes(:votes)` to avoid extra db queries

Reply:
👍 done